src/17D6A0.c

```diff
@@ -1,21 +1,101 @@
 #include "common.h"
+#include "effects.h"
 
+extern PopupMessage popupMessages[32];
+extern PopupMessage* D_802838F8;
 extern s16 D_8029F64C;
+extern s16 D_8029F640;
+extern s16 D_8029F64A;
+extern s16 D_8029F64E;
+extern s16 D_8029F650;
+extern void func_8024FB3C(PopupMessage* popup);
+void btl_show_message_popup(PopupMessage* popup);
+
+void func_8024EDC0(void) {
+    PopupMessage* popup;
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(popupMessages); i++) {
+        popup = &popupMessages[i];
+        popup->active = FALSE;
+        popup->message = NULL;
+    }
+}
+
+void func_8024EDEC(void) {
+    PopupMessage* popup;
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(popupMessages); i++) {
+        popup = &popupMessages[i];
```

Comment:
IDM particularly but in my opinion its nicer to declare popup on this line also instead of at the top of the function

Reply:
This goes for all funcs here